.github/workflows/kustomize-build-ci.yaml

```diff
@@ -0,0 +1,58 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: kustomize-build-ci
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'kustomize/**'
+      - '.github/workflows/kustomize-build-ci.yml'
+  pull_request:
+    paths:
+      - 'kustomize/**'
+      - '.github/workflows/kustomize-build-ci.yml'
+jobs:
+  job:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+      - name: kustomize build base
+        run: |
+          cd kustomize/
+          kubectl kustomize .
+      - name: kustomize build each component
+        run: |
+          cd kustomize/
+          COMPONENTS=$(ls components/)
+          for c in $COMPONENTS;
+          do
+              echo "## kustomize build for " + $c
+              rm kustomization.yaml
+              kustomize create --resources base
+              kustomize edit add component components/$c
+              kustomize build .
+          done
+      - name: kustomize build all components
+        run: |
+          cd kustomize/
+          rm kustomization.yaml
+          kustomize create --resources base
+          COMPONENTS=$(ls components/)
+          for c in $COMPONENTS;
+          do
+              kustomize edit add component components/$c
+          done
+          kustomize build .
```

Comment:
**Suggestion:** I don't think "kustomize build all components" is a CUJ that we need to support. Once we add Spanner as a Kustomize component, it doesn't make sense to use both the `spanner` component and `memorystore` together. Ket's remove this test.

Reply:
That's a good point about the "conflict" between `memorystore` and the coming `spanner` one, we don't want both at the same time/deployment.

But I still think that we need to unit test a combination of multiple Kustomize components, that's exactly where we could have issues/conflicts. So I think I will "hard-code" a scenario of comibining multiple components (that's a CUJ we advertise in our `kustomize/README.md` doc ;)). And we will see later if we could automate something around that, why not with a `kustomize-cujs.config|json` file for example... let's see that later when will have more and more `components`.

Reply:
Ahh, that totally makes sense.
We should definitely test at least one combination (since we advertise support for multiple).
I agree with hardcoding a single scenario for now.

Happy to explore introducing `kustomize-cujs.config|json` file in the future — as long as we're not testing every possible combination (which can easily grow to 1000s 😅 ).

Reply:
Just did this custom cuj: https://github.com/GoogleCloudPlatform/microservices-demo/pull/1097/commits/a4c8e39bd26edad305e0ade46780d6309b13f26c

Reply:
Looks good to me!
I'm also open to excluding `kustomize edit remove resource network-policy-redis.yaml`. But totally up to you.
Resolved!